include/zephyr/drivers/sensor/battery.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 Embeint Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENSOR_BATTERY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SENSOR_BATTERY_H_
+
+#include <stdint.h>
+#include <errno.h>
+
+#include <zephyr/devicetree.h>
+#include <zephyr/math/interpolation.h>
+#include <zephyr/sys/util_macro.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief battery API
+ * @defgroup battery_apis battery APIs
+ * @{
+ */
+
+/* Battery chemistry enumeration.
+ * Value names must match those from dts/bindings/battery.yaml
+ */
+enum battery_chemistry {
+	BATTERY_CHEMISTRY_UNKNOWN = 0,
+	BATTERY_CHEMISTRY_NICKEL_CADMIUM,
+	BATTERY_CHEMISTRY_NICKEL_METAL_HYDRIDE,
+	BATTERY_CHEMISTRY_LITHIUM_ION,
+	BATTERY_CHEMISTRY_LITHIUM_ION_POLYMER,
+	BATTERY_CHEMISTRY_LITHIUM_ION_IRON_PHOSPHATE,
+	BATTERY_CHEMISTRY_LITHIUM_ION_MANGANESE_OXIDE,
+};
+
+/* Length of open circuit voltage table */
+#define BATTERY_OCV_TABLE_LEN 11
+
+/**
+ * @brief Get the battery chemistry enum value
+ *
+ * @param node_id node identifier
+ */
+#define BATTERY_CHEMISTRY_DT_GET(node_id)                                                          \
+	UTIL_CAT(BATTERY_CHEMISTRY_, DT_STRING_UPPER_TOKEN_OR(node_id, device_chemistry, UNKNOWN))
```

Comment:
hey is this used anywhere? is it really needed?

Reply:
Historically I have used this in hardware test applications to select acceptable voltage ranges.
I.e. if LiFePO4 3-3.6V is fine, LiPo, 3.7-4.2V is fine.
It's not strictly required, but I have found it useful.

Reply:
I mean the `device-chemestry` property specifically, I don't see any reference to `BATTERY_CHEMISTRY_DT_GET`, what am I missing?

Reply:
That is what I mean, I use the `device-chemistry` property in some test applications, but not general library code